### DISCUSSION TITLE
Using Guzzle Client in Behat tests

### DISCUSSION THREAD

**Discussion by franiglesias** [dunglas/symfony-docker#626] [Q&A] [answered]
Hi.

I want to write behat tests to develop my app. I need to connect using Guzzle client from inside the container, so the tests could access to the api endpoints.

How do you configure guzzle. I've tried with the cert option, like this:

`
    public function iVisitWebRoot()
    {
        $client = new \GuzzleHttp\Client();
        $this->request = $client->get( 
            'https://app.localhost/', 
            ['cert' => '/data/caddy/certificates/local/localhost/localhost.crt']);
    }
`

but I get this error:

`cURL error 58: unable to set private key file: '/data/caddy/certificates/local/localhost/localhost.crt' type PEM (see https://curl.haxx.se/libcurl/c/libcurl-errors.html) for https://app.localhost/`

Any guidance would be greatly appreciated. Thanks,

**Comment by 7-zete-7** [comment]
Hello @franiglesias!

Are you sure you have the `/data/caddy/certificates/local/localhost/localhost.crt` file in your container?

Can you run the following commands to make sure the file is there and in its format?

```shell
ls -lah /data/caddy/certificates/local/localhost/localhost.crt
ls -lah /data/caddy/certificates/local/localhost/
head -n 3 /data/caddy/certificates/local/localhost/localhost.crt
```

**Reply by franiglesias** [reply, marked as answer]
Thanks @7-zete-7 

Here is the output in the container. The file is there... but I didn't need it.

```
❯ make bash
root@2314c8d4eb51:/app# ls -lah /data/caddy/certificates/local/localhost/localhost.crt
-rw------- 1 root root 1.4K May  2 06:52 /data/caddy/certificates/local/localhost/localhost.crt
root@2314c8d4eb51:/app# ls -lah /data/caddy/certificates/local/localhost/
total 20K
drwx------ 2 root root 4.0K Apr 27 17:11 .
drwx------ 3 root root 4.0K Apr 27 17:11 ..
-rw------- 1 root root 1.4K May  2 06:52 localhost.crt
-rw------- 1 root root   31 May  2 06:52 localhost.json
-rw------- 1 root root  227 May  2 06:52 localhost.key
root@2314c8d4eb51:/app# head -n 3 /data/caddy/certificates/local/localhost/localhost.crt
-----BEGIN CERTIFICATE-----
MIIBvjCCAWOgAwIBAgIQDbl/ueV13HgCmUfvFtjPUjAKBggqhkjOPQQDAjAzMTEw
LwYDVQQDEyhDYWRkeSBMb2NhbCBBdXRob3JpdHkgLSBFQ0MgSW50ZXJtZWRpYXRl
```

But I've started to make some checks using curl inside the container and found two things I was doing wrong:

* I was using the wrong url (it should be https://localhost (not app.localhost... )
* Symfony default page returns 404 status code. That confused me until I retrieved it with curl and understood that the output was that welcome page

All my fault! 🤦‍♂️ 

Thank you a lot for answering.
